File: facade/python/facade.py
from subsystems import CoffeeMachine, Delivery, Deposit


class CoffeeShop:
    def __init__(self) -> None:
        self.coffeeMachine = CoffeeMachine()
        self.delivery = Delivery()
        self.deposit = Deposit()

    def order(self) -> str:
        self.deposit.getCoffee()
        self.coffeeMachine.on()
        self.coffeeMachine.prepare()
        self.delivery.send()

File: javascript/facade/facade.ts
import { CoffeeMachine, Delivery, Deposit } from "./subsystems";

export class CoffeeShop {
    order() {
        const deposit = new Deposit()
        const coffeeMachine = new CoffeeMachine()
        const delivery = new Delivery()

        deposit.getCoffee()
        coffeeMachine.on()
        coffeeMachine.prepare()
        delivery.send()
    }
}